include/log4cplus/spi/filter.h

```diff
@@ -303,6 +303,57 @@ namespace log4cplus {
             Function function;
         };
 
+        class LOG4CPLUS_EXPORT NDCMatchFilter : public Filter
+        {
+            public:
+              // ctors
+                NDCMatchFilter();
+                NDCMatchFilter(const log4cplus::helpers::Properties& p);
+
+                /**
+                 * Returns {@link #NEUTRAL} is there is no string match.
+                 */
+                virtual FilterResult decide(const InternalLoggingEvent& event) const;
+
+            private:
+              // Methods
+                LOG4CPLUS_PRIVATE void init();
+
+              // Data
+                /** Do we return ACCEPT when a match occurs. Default is <code>true</code>. */
+                bool acceptOnMatch;
+                /** return NEUTRAL if NDC::get() is empty or ndcToMatch is empty. Default is <code>true</code>. */
+                bool neutralOnEmpty;
+                log4cplus::tstring ndcToMatch;
+        };
+
+        class LOG4CPLUS_EXPORT MDCMatchFilter : public Filter
```

Comment:
Same here. Could you please add some Doxygen documentation describing the filter?